sensorium/training/trainers.py

```diff
@@ -109,7 +160,17 @@ def full_objective(model, dataloader, data_key, *args, **kwargs):
 
     n_iterations = len(LongCycler(dataloaders["train"]))
 
-    optimizer = torch.optim.Adam(model.parameters(), lr=lr_init)
+    if include_kldivergence:
+        cluster_centers = torch.zeros(
+            cluster_number, 128, dtype=torch.float, requires_grad=True
```

Comment:
I guess the 128 comes from the dimensionality of the readouts. It might be better if you do do not hardcode it but dynamically assign it based on the model

Reply:
True. I will fix that